acceptance-tests/src/test/java/tech/pegasys/web3signer/tests/signing/BlsSigningAcceptanceTest.java

```diff
@@ -113,28 +116,90 @@ public void ableToSignUsingAzure() {
     metadataFileHelpers.createAzureYamlFileAt(
         keyConfigFile, clientId, clientSecret, tenantId, keyVaultName, secretName);
 
-    signAndVerifySignature();
+    signAndVerifySignature(ArtifactType.BLOCK);
   }
 
   @Test
-  public void ableToSignUsingYubiHsm() {
+  public void ableToSignUsingYubiHsm() throws JsonProcessingException {
     final Path configFile = testDirectory.resolve("yubihsm_1.yaml");
     metadataFileHelpers.createYubiHsmYamlFileAt(configFile, KeyType.BLS);
 
-    signAndVerifySignature(yubiHsmShellEnvMap());
+    signAndVerifySignature(ArtifactType.BLOCK, yubiHsmShellEnvMap());
+  }
+
+  @Test
+  public void failsIfSigningRootDoesNotMatchSigningData() throws JsonProcessingException {
+    final String configFilename = publicKey.toString().substring(2);
+
+    final Path keyConfigFile = testDirectory.resolve(configFilename + ".yaml");
+    metadataFileHelpers.createKeyStoreYamlFileAt(keyConfigFile, keyPair, KdfFunction.SCRYPT);
+
+    setupSigner("eth2", null);
+
+    final Eth2SigningRequestBody request = Eth2RequestUtils.createBlockRequest();
+    final Eth2SigningRequestBody requestWithMismatchedSigningRoot =
+        new Eth2SigningRequestBody(
+            request.getType(),
+            Bytes32.ZERO,
+            request.getForkInfo(),
+            request.getBlock(),
+            request.getAttestation(),
+            request.getAggregationSlot(),
+            request.getAggregateAndProof(),
+            request.getVoluntaryExit(),
+            request.getRandaoReveal(),
+            request.getDeposit());
+
+    final Response response =
+        signer.eth2Sign(keyPair.getPublicKey().toString(), requestWithMismatchedSigningRoot);
+    assertThat(response.getStatusCode()).isEqualTo(500);
+  }
+
+  @Test
+  public void ableToSignWithoutSigningRootField() throws JsonProcessingException {
```

Comment:
It's required by the spec. But we only use it compare with the computed signing root. Made it optional so it would make the API easier to use in some cases e.g. signing the deposit data which can't be done via Web3Signer with Teku. As I don't there is much value is specifying other than it's in the spec. Not too concerned if we make it required again, so if you feel strongly about it I can make it mandatory again.